cluster-http/src/main/scala/akka/management/cluster/scaladsl/ClusterHttpManagementRoutes.scala

```diff
@@ -113,6 +113,108 @@ object ClusterHttpManagementRoutes extends ClusterHttpManagementJsonProtocol {
     }
   }
 
+  private def routeGetClusterDomainEvents(cluster: Cluster) = {
+    import akka.actor.ActorRef
+    import akka.cluster.ClusterEvent
+    import akka.http.scaladsl.marshalling.sse.EventStreamMarshalling._
+    import akka.http.scaladsl.model.sse.ServerSentEvent
+    import akka.stream.{ Materializer, OverflowStrategy }
+    import akka.stream.scaladsl.Source
+    import scala.concurrent.{ ExecutionContext, Promise }
+
+    val eventClasses: Map[String, Class[_]] = Map(
+      "ClusterDomainEvent" -> classOf[ClusterEvent.ClusterDomainEvent],
+      "MemberEvent" -> classOf[ClusterEvent.MemberEvent],
+      "MemberJoined" -> classOf[ClusterEvent.MemberJoined],
+      "MemberWeaklyUp" -> classOf[ClusterEvent.MemberWeaklyUp],
+      "MemberUp" -> classOf[ClusterEvent.MemberUp],
+      "MemberLeft" -> classOf[ClusterEvent.MemberLeft],
+      "MemberExited" -> classOf[ClusterEvent.MemberExited],
+      "MemberDowned" -> classOf[ClusterEvent.MemberDowned],
+      "MemberRemoved" -> classOf[ClusterEvent.MemberRemoved],
+      "LeaderChanged" -> classOf[ClusterEvent.LeaderChanged],
+      "RoleLeaderChanged" -> classOf[ClusterEvent.RoleLeaderChanged],
+      "ClusterShuttingDown" -> ClusterEvent.ClusterShuttingDown.getClass,
+      "ReachabilityEvent" -> classOf[ClusterEvent.ReachabilityEvent],
+      "UnreachableMember" -> classOf[ClusterEvent.UnreachableMember],
+      "ReachableMember" -> classOf[ClusterEvent.ReachableMember],
+      "DataCenterReachabilityEvent" -> classOf[ClusterEvent.DataCenterReachabilityEvent],
+      "UnreachableDataCenter" -> classOf[ClusterEvent.UnreachableDataCenter],
+      "ReachableDataCenter" -> classOf[ClusterEvent.ReachableDataCenter]
+    )
+
+    extractMaterializer { implicit mat: Materializer =>
+      implicit val ec: ExecutionContext = mat.executionContext
+
+      get {
+        parameter("type".as[String].*) { providedEventTypes =>
+          val classes =
+            if (providedEventTypes.nonEmpty)
+              providedEventTypes.foldLeft(List.empty[Class[_]]) {
+                case (accum, eventType) =>
+                  eventClasses.get(eventType).toList ::: accum
+              }
+            else
+              List(classOf[ClusterEvent.ClusterDomainEvent])
+
+          val eventualActorRef = Promise[Option[ActorRef]]
```

Comment:
I think this could also be a `Promise[ActorRef]`, which we could `tryFailure` when the stream completes before successfully completing the promise. This is okay as well, though.

Reply:
Yeah, I see that's nicer in a way then since success doesn't have the option to deal with. I don't like having to pick an exception type though (probably should be `NoStackTrace`), so rather leave it as-is.